pkg/keys/in_memory.go

```diff
@@ -43,42 +49,68 @@ func NewInMemory(ctx context.Context) (*InMemory, error) {
 	}, nil
 }
 
-// AddSigningKey generates a new ECDSA P256 Signing Key identified by
+// CreateSigningKey generates a new ECDSA P256 Signing Key identified by
 // the provided keyID
-func (k *InMemory) AddSigningKey(keyID string) error {
+func (k *InMemory) CreateSigningKey(keyID string) (*ecdsa.PrivateKey, error) {
 	k.mu.Lock()
 	defer k.mu.Unlock()
 
 	if _, ok := k.signingKeys[keyID]; ok {
-		return fmt.Errorf("key already exists: %v", keyID)
+		return nil, fmt.Errorf("key already exists: %v", keyID)
 	}
 
 	pk, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
 	if err != nil {
-		return fmt.Errorf("unable to generate private key: %w", err)
+		return nil, fmt.Errorf("unable to generate private key: %w", err)
+	}
+
+	k.signingKeys[keyID] = pk
+	return pk, nil
+}
+
+// AddSigningKey adds a new ECDSA P256 Signing Key identified by the provided
+// keyID.
+func (k *InMemory) AddSigningKey(keyID string, pk *ecdsa.PrivateKey) error {
+	k.mu.Lock()
+	defer k.mu.Unlock()
+
+	if _, ok := k.signingKeys[keyID]; ok {
+		return fmt.Errorf("key already exists: %v", keyID)
 	}
 
 	k.signingKeys[keyID] = pk
 	return nil
 }
 
-// AddEncryptionKey generates a new encryption key identified by
-// the provided keyID.
-func (k *InMemory) AddEncryptionKey(keyID string) error {
+// CreateEncryptionKey generates and stores new encryption key identified by the
+// provided keyID.
+func (k *InMemory) CreateEncryptionKey(keyID string) ([]byte, error) {
 	k.mu.Lock()
 	defer k.mu.Unlock()
 
 	if _, ok := k.cryptoKeys[keyID]; ok {
-		return fmt.Errorf("key already exists: %v", keyID)
+		return nil, fmt.Errorf("key already exists: %v", keyID)
 	}
 
 	key := make([]byte, 32)
 	if _, err := io.ReadFull(rand.Reader, key); err != nil {
-		return fmt.Errorf("failed to read random bytes: %w", err)
+		return nil, fmt.Errorf("failed to read random bytes: %w", err)
 	}
 
 	k.cryptoKeys[keyID] = key
+	return key, nil
+}
+
+// AddEncryptionKey stores the key on the system.
+func (k *InMemory) AddEncryptionKey(keyID string, key []byte) error {
+	k.mu.Lock()
+	defer k.mu.Unlock()
 
+	if _, ok := k.cryptoKeys[keyID]; ok {
+		return fmt.Errorf("key already exists: %v", keyID)
```

Comment:
%q?

Reply:
`%v` is "best for the thing (which is just the string value). `%q` would quote the string, which isn't desirable here. You could argue for `%s`, which is more technically correct, but this is only used for testing anyway, so I'm not super worried 😄

Reply:
make sense
/unhold